core/src/main/scala/akka/persistence/cassandra/query/scaladsl/CassandraReadJournal.scala

```diff
@@ -434,11 +434,10 @@ class CassandraReadJournal(system: ExtendedActorSystem, cfg: Config)
       queryPluginConfig.fetchSize,
       Some(queryPluginConfig.refreshInterval),
       s"eventsByPersistenceId-$persistenceId",
-      extractor = Extractors.persistentRepr
+      extractor = Extractors.taggedPersistentRepr
     )
       .mapMaterializedValue(_ => NotUsed)
-      .map(p => mapEvent(p.persistentRepr))
-      .mapConcat(r => toEventEnvelopes(r, r.sequenceNr))
+      .mapConcat(r => toEventEnvelopes(mapEvent(r.pr), TimeBasedUUID(r.offset)))
```

Comment:
This is not something we can just change like this. It would break existing usage and the `eventsByPersistenceId` is supposed to use the sequence number as the offset.

What we can do is to add another query (specific for akka-persistence-cassandra) that returns another EventEnvelope type that includes the TimeBasedUUID. It should probably include other things as well, sequence number and also tags can be useful.